### PR TITLE
Fix issues with extra backslashes in path string

### PR DESCRIPTION
**Pull request by jamacku**
From `/abc//def/` to `/abc/def/`

Resolve: #1485304

### CODE REVIEW

File: alternatives.c [hunk 1]
```diff
@@ -255,6 +255,27 @@ char *parseLine(char **buf) {
...
+  while (*pr) {
+    *pw = *pr;
+    *pr++;
+    if ((*pw == c) && (state == 0)) {
+      pw++;
+      state = 1;
+    } else if (*pw != c) {
+      pw++;
+      state = 0;
+    } else {
```
Comment (lnykryn) [inline alternatives.c]: Remove the empty else.

File: alternatives.c [hunk 2]
```diff
@@ -255,6 +255,27 @@ char *parseLine(char **buf) {
     return strdup(start);
 }
 
+void clean_path(char *path, char c) {
```
Comment (lnykryn) [inline alternatives.c]: I can't imagine that we would like to remove anything else than /. I would drop the char and hardcode '/'

File: alternatives.c (hunk 1 above)
Comment (jamacku) [reply alternatives.c]: Done

File: alternatives.c (hunk 2 above)
Comment (jamacku) [reply alternatives.c]: True, thank you.

**Comment by jamacku** [comment]
> Also, it looks that the code around uses four spaces indentation, while this is written with only two.

Thank you, I totally overlooked it.

**Comment by lnykryn** [comment]
Looks good to me, tested with:
for i in  "aaa/bbb////ccc" "aaa/bbb/ccc" "aaa/bbb/ccc"  "//aaa/.////ccc" "/aaa/./ccc" "/aaa/ccc"  "///" "/" "/"  "///.//" "/." "/"  "///.//.///" "/./." "/"  "////.././///../." "/.././../." "/../.."  "." "." "."  "./" "." "."  ".///.//./." "./././." "."  ".///.//././/" "./././." "."  "//./aaa///.//./.bbb/..///c.//d.dd///..eeee/."  "/./aaa/././.bbb/../c./d.dd/..eeee/."  "/aaa/.bbb/../c./d.dd/..eeee"  "//./aaa///.//./.bbb/..///c.//d.dd///..eeee/.."  "/./aaa/././.bbb/../c./d.dd/..eeee/.."  "/aaa/.bbb/../c./d.dd/..eeee/.."  ".//./aaa///.//./.bbb/..///c.//d.dd///..eeee/.."  "././aaa/././.bbb/../c./d.dd/..eeee/.."  "aaa/.bbb/../c./d.dd/..eeee/.."  "..//./aaa///.//./.bbb/..///c.//d.dd///..eeee/.."  ".././aaa/././.bbb/../c./d.dd/..eeee/.."  "../aaa/.bbb/../c./d.dd/..eeee/.." ; do a=$(echo -n "$i" | sed "s#/\+#/#g") ; b=$(./a.out "$i"); echo "$i -> $a $b $([ $a = $b ] && echo -n ok )"  ; done

**Pull request merged**